build/build.go

```diff
@@ -525,6 +538,18 @@ func NewSession(options *Options) (*Session, error) {
 	return s, nil
 }
 
+func (s *Session) CheckMod(pkg *PackageData) {
+	if pkg.Goroot {
+		s.mod = nil
+	} else {
+		s.mod, _ = fastmod.LoadPackage(pkg.Dir, s.bctx)
+	}
+}
+
+func (s *Session) CheckModFromDir(dir string) {
+	s.mod, _ = fastmod.LoadPackage(dir, s.bctx)
+}
```

Comment:
I rather like this PR, but these two functions make the code rely on side-effects, which is quite error prone and brittle. Perhaps module info should be added to PackageData?

Note: I'm not a GopherJS maintainer, so my words carry limited weight :)

Reply:
https://github.com/gopherjs/gopherjs/pull/971/commits/246fd7500e264213537cdcd26a2cbde0b61ded59 
simplify mod check and clear code.